dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -27,23 +29,26 @@ abstract class TypedEncoder[T](implicit val classTag: ClassTag[T]) extends Seria
 //   - import TypedEncoder.usingInjection
 //   - import TypedEncoder.usingDerivation
 // """)
-object TypedEncoder {
+object TypedEncoder extends TypedEncoder0 {
```

Comment:
Is `TypedEncoder0` a `LowPriorityTypedEncoder`? Why do we need prioritization here?